src/OpenSpaceToolkit/Astrodynamics/Access/Generator.cpp

```diff
@@ -311,6 +551,171 @@ Generator Generator::AerMask(
     return {anEnvironment, aerFilter};
 }
 
+Array<Access> Generator::generateAccessesFromIntervals(
+    const Array<physics::time::Interval>& someIntervals,
+    const physics::time::Interval& anInterval,
+    const Trajectory& aFromTrajectory,
+    const Trajectory& aToTrajectory
+) const
+{
+    const Shared<const Celestial> earthSPtr = this->environment_.accessCelestialObjectWithName("Earth");
+
+    return someIntervals
+        .map<Access>(
+            [&anInterval, &aFromTrajectory, &aToTrajectory, &earthSPtr, this](
+                const physics::time::Interval& anAccessInterval
+            ) -> Access
+            {
+                return Generator::GenerateAccess(
+                    anAccessInterval, anInterval, aFromTrajectory, aToTrajectory, earthSPtr, this->tolerance_
+                );
+            }
+        )
+        .getWhere(
+            [this](const Access& anAccess) -> bool
+            {
+                return this->accessFilter_ ? this->accessFilter_(anAccess) : true;
+            }
+        );
+}
+
+Array<physics::time::Interval> Generator::computePreciseCrossings(
```

Comment:
Regarding your choice to use a separate function to sort of "redo" the interval computation math more precisely here:
- On first glance, it seems like some of logic here is redone after it is already done once above in the main body of the `computeAccessesWithGroundTargets` function. Could you not have instead included a call to a `computePreciseCrossing` (without the s) to refine the crossing estimate within the main computation loop above? Instead of going once through all the access intervals for the coarse calculation and then again looping through them all for the fine calculation?

Reply:
You can, but then you:
- Lose the ability to just do the coarse computation and bail early, which might be suitable for a lot of cases (for example imaging window computations), since the precise crossings actually takes a majority of the computation time.

Also none of the logic is repeated, it's just more memory efficient to do it in one loop but quite neglibily, and the performance is also almost identical.